File: TugasModul4/src/magnitude_client.cpp
#include "ros/ros.h"
#include "package1/magnitude.h"

int main(int argc, char **argv){
    // Inisialisasi 
    ros::init(argc,argv,"magnitude_client");
    
    // Inisialisasi Handler Node
    ros::NodeHandle nh;

    // Inisialisasi client dari handler
    ros::ServiceClient client = nh.serviceClient<package1::magnitude>("magnitude");

    // Rate pengiriman pesan sebesar 1hz (1 pesan setiap detik)
    ros::Rate loop_rate(1);

    while(ros::ok()) {
        // Inisialisasi variabel sesuai tipe data srv
        package1::magnitude data;

        // Memberi nilai kepada seluruh data request
        data.request.komponenX = 12;
        data.request.komponenY = 24;
        data.request.komponenZ = 36;

        if (client.call(data)){
            ROS_INFO("[Client] Magnitude yang dicari adalah: %lf", data.response.magnitude);
        }
        else {
            ROS_INFO("[Client] Service gagal dipanggil");
        }

        ros::spinOnce();
        loop_rate.sleep();
    }

}

File: TugasModul4/TPD_4/src/package1/script/magnitude_client.py
#!/usr/bin/env python3

import rospy
from package1.srv import magnitude, magnitudeResponse

def magnitude_client(x, y, z):
    # Inisialisasi Node
    rospy.init_node("magnitude_client")

    rospy.wait_for_service("magnitude_service")
    rate = rospy.Rate(1)

    while not rospy.is_shutdown():
        try:
            # Inisialisasi Client
            findMagnitude = rospy.ServiceProxy('magnitude_service', magnitude)
            response = findMagnitude(x, y, z)

            # Mencetak hasil response
            rospy.loginfo("[Client] Magnitude yang dicari: %lf" % response.magnitude)
            rate.sleep()
        except rospy.ServiceException as e:
            print("[Client] Service call failed %s", e)

if __name__ == '__main__':
    try:
        magnitude_client(12, 12, 12)
    except rospy.ROSInterruptException:
        pass